scripts/ensure_s3_directories.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+# Script to ensure S3 directories exist for Spark job logs
+# This script is called from setup.sh when SPARK_JOB_LOG_DIR is set
+
+if [ -z "$SPARK_JOB_LOG_DIR" ]; then
+    echo "SPARK_JOB_LOG_DIR not set, skipping directory creation"
+    exit 0
+fi
+
+if [ -z "$MINIO_URL" ] || [ -z "$MINIO_LOG_USER_ACCESS_KEY" ] || [ -z "$MINIO_LOG_USER_SECRET_KEY" ]; then
+    echo "MinIO credentials not set, exiting"
+    exit 1
+fi
+
+# Set MinIO client config directory to a writable location
+export MC_CONFIG_DIR="/tmp/.mc"
+mkdir -p "$MC_CONFIG_DIR"
+
+if [ -z "$SPARK_JOB_LOG_DIR_CATEGORY" ]; then
+    echo "SPARK_JOB_LOG_DIR_CATEGORY not set"
+    FULL_S3_PATH="$SPARK_JOB_LOG_DIR"
+    SPARK_JOB_LOG_DIR_CATEGORY="default"
+else
+    FULL_S3_PATH="${SPARK_JOB_LOG_DIR}/${SPARK_JOB_LOG_DIR_CATEGORY}"
+fi
+S3_URL_WITHOUT_PROTOCOL=${FULL_S3_PATH#s3a://}
```

Comment:
This name is a bit off, it's really the S3 path

Reply:
👍